plugins/samples/redirect_bulk/plugin.cc

```diff
@@ -0,0 +1,129 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START serviceextensions_plugin_redirect_bulk]
+#include "proxy_wasm_intrinsics.h"
+#include "absl/strings/ascii.h"
+#include "absl/strings/str_split.h"
+#include "absl/strings/str_cat.h"
+#include <unordered_map>
+#include <string>
+
+// Root context class that handles plugin configuration and domain mappings.
+class MyRootContext : public RootContext {
+public:
+  explicit MyRootContext(uint32_t id, std::string_view root_id)
+      : RootContext(id, root_id) {}
+
+  // Called when the plugin is configured with the provided configuration.
+  bool onConfigure(size_t config_len) override {
+    // Retrieve the plugin configuration data.
+    auto config_data = getBufferBytes(WasmBufferType::PluginConfiguration, 0, config_len);
+    
+    // If no configuration is provided, log a warning and continue.
+    if (!config_data || config_data->size() == 0) {
+      LOG_WARN("No configuration provided, no redirects will be performed");
+      return true;
+    }
+
+    // Convert the configuration data to a string.
+    std::string config_str = config_data->toString();
+    // Split the configuration string into lines.
+    std::vector<std::string> lines = absl::StrSplit(config_str, '\n');
+
+    // Parse each line to extract domain mappings.
+    for (const auto& line : lines) {
+      // Strip whitespace and skip empty lines or comments.
+      absl::string_view stripped = absl::StripAsciiWhitespace(line);
+      if (stripped.empty() || stripped[0] == '#') continue;
+
+      // Split the line into source and target domains.
+      std::vector<absl::string_view> parts = 
+          absl::StrSplit(stripped, absl::MaxSplits(' ', 1));
```

Comment:
I think this probably should *not* specify absl::MaxSplits(), so that the parts.size() check  a few lines later can detect the case where config contains too many items on a line.